src/expr/src/expr/expr_in.rs

```diff
@@ -34,6 +34,8 @@ pub(crate) struct InExpression {
     return_type: DataType,
 }
 
+// This is a False Positive listed in https://rust-lang.github.io/rust-clippy/master/index.html#mutable_key_type.
+#[allow(clippy::mutable_key_type)]
```

Comment:
this is becasue we add `ScalarImpl::Bytea::(Bytes)`, and they think `Bytes` has interior mutability that may hurts `Hash` or `Ord`. 

But they also say this case is a False Positive but due to compiler problems they have no ability to recognize it. So I add this clippy::allow to make it pass.

Reply:
This may introduce Overhead that everytime we create `HashSet` ot `HashMap` that contains `Datum` as key, it will complain on this rule and we need to add this `allow` manually. 

Another is to replace `Bytes` with `Vec<u8>` inside `ScalarImpl::Bytea`. Or we add this rule.. Looking to suggestions.

Reply:
Perfer `Vec<u8>` over `Bytes`. IIUC, `Bytes` is designed for efficiently manipulating network buffer, so it contains a lot of methods for splitting, concating, etc, which we don't need.

Reply:
Fixed

Reply:
Can we remove this line?